File: server/internal/model/ticket.go
package model

import "time"

const (
	TicketStatusPending    = "pending"
	TicketStatusInProgress = "in_progress"
	TicketStatusCompleted  = "completed"
)

const (
	TicketPriorityLow    = "low"
	TicketPriorityMedium = "medium"
	TicketPriorityHigh   = "high"
)

type Ticket struct {
	BaseModel
	UserID      uint       `gorm:"not null;index" json:"user_id"`
	CategoryID  uint       `gorm:"not null;index" json:"category_id"`
	Title       string     `gorm:"size:200;not null" json:"title"`
	Description string     `gorm:"type:text;not null" json:"description"`
	Status      string     `gorm:"size:20;not null;default:'pending';index" json:"status"`
	Priority    string     `gorm:"size:20;not null;default:'medium'" json:"priority"`
	HandledBy   *uint      `gorm:"index" json:"handled_by,omitempty"`
	HandledAt   *time.Time `json:"handled_at,omitempty"`
	ClosedAt    *time.Time `json:"closed_at,omitempty"`
}

func (Ticket) TableName() string {
	return "ticket"
}

type TicketCategory struct {
	BaseModel
	Name        string `gorm:"size:50;not null;uniqueIndex" json:"name"`
	NameEN      string `gorm:"size:50;not null;uniqueIndex" json:"name_en"`
	Description string `gorm:"size:200" json:"description"`
	SortOrder   int    `gorm:"not null;default:0" json:"sort_order"`
	Enabled     bool   `gorm:"not null;default:true" json:"enabled"`
}

func (TicketCategory) TableName() string {
	return "ticket_category"
}

type TicketReply struct {
	BaseModel
	TicketID   uint   `gorm:"not null;index" json:"ticket_id"`
	UserID     uint   `gorm:"not null;index" json:"user_id"`
	Content    string `gorm:"type:text;not null" json:"content"`
	IsInternal bool   `gorm:"not null;default:false" json:"is_internal"`
}

func (TicketReply) TableName() string {
	return "ticket_reply"
}

type TicketStatusHistory struct {
	ID         uint      `gorm:"primarykey" json:"id"`
	TicketID   uint      `gorm:"not null;index" json:"ticket_id"`
	FromStatus string    `gorm:"size:20" json:"from_status"`
	ToStatus   string    `gorm:"size:20;not null" json:"to_status"`
	ChangedBy  uint      `gorm:"not null" json:"changed_by"`
	ChangedAt  time.Time `gorm:"autoCreateTime" json:"changed_at"`
}

func (TicketStatusHistory) TableName() string {
	return "ticket_status_history"
}

File: static-react/src/types/api/ticket.ts
import type { CommonSearchParams, PaginatedResponse } from '@/types/api/common'

export type TicketStatus = 'pending' | 'in_progress' | 'completed'
export type TicketPriority = 'low' | 'medium' | 'high'

export interface TicketItem {
  id: number
  user_id: number
  category_id: number
  title: string
  description: string
  status: TicketStatus
  priority: TicketPriority
  handled_by?: number
  handled_at?: string
  closed_at?: string
  created_at: string
  updated_at: string
}

export interface TicketCategory {
  id: number
  name: string
  name_en: string
  description: string
  sort_order: number
  enabled: boolean
  created_at: string
  updated_at: string
}

export interface TicketReply {
  id: number
  ticket_id: number
  user_id: number
  content: string
  is_internal: boolean
  created_at: string
  updated_at: string
}

export interface TicketListParams extends Partial<CommonSearchParams> {
  status?: TicketStatus | ''
}

export interface AdminTicketListParams extends Partial<CommonSearchParams> {
  status?: TicketStatus | ''
  keyword?: string
  category_id?: number
  user_id?: number
}

export interface CreateTicketParams {
  category_id: number
  title: string
  description: string
  priority?: TicketPriority
}

export interface AddReplyParams {
  content: string
}

export interface AdminAddReplyParams extends AddReplyParams {
  is_internal?: boolean
}

export interface UpdateStatusParams {
  status: TicketStatus
}

export interface UpdatePriorityParams {
  priority: TicketPriority
}

export interface UpsertCategoryParams {
  name: string
  name_en: string
  description?: string
  sort_order?: number
  enabled?: boolean
}

export interface TicketStatusHistory {
  id: number
  ticket_id: number
  from_status: string
  to_status: TicketStatus
  changed_by: number
  changed_at: string
}

export interface Statistics {
  total: number
  status: Record<TicketStatus, number>
  category: Record<string, number>
  recent_7d: number
  recent_30d: number
  pendingCount: number
}

export type TicketListResponse = PaginatedResponse<TicketItem>
